### ISSUE TITLE
False positive "invalid base for `typing.NewType`" with `float`

### ISSUE THREAD

**Issue by LeeeeT** [astral-sh/ty#1818]
### Summary

```python
from typing import NewType

MyFloat = NewType('MyFloat', float)
```

```
error[invalid-newtype]: invalid base for `typing.NewType`
 --> file.py:3:30
  |
1 | from typing import NewType
2 |
3 | MyFloat = NewType('MyFloat', float)
  |                              ^^^^^ type `int | float`
  |
info: The base of a `NewType` must be a class type or another `NewType`.
info: rule `invalid-newtype` is enabled by default
```

### Version

ty 0.0.1-alpha.32 (84a188116 2025-12-05)

**Comment by carljm** [comment]
@oconnor663 can you look into this? I think `NewType` will need special handling for `float` and `complex`. This should work:

```py
from typing import NewType

MyFloat = NewType('MyFloat', float)

def f(x: MyFloat):
    ...

f(MyFloat(1))
f(MyFloat(1.0))
```

**Comment by MichaReiser** [comment]
Do we need the same special casing in other places like base classes etc too?

**Comment by AlexWaygood** [comment]
> Do we need the same special casing in other places like base classes etc too?

I don't think so — class bases are parsed as value expressions. The reason why we infer a union here is because the second argument to NewType is parsed as a type expression.

**Comment by MichaReiser** [comment]
ah right, I always forget this distinction. 

We have this crime in `ty_ide`. I hope there's a better way to handle `float` in `NewType` inference

https://github.com/astral-sh/ruff/blob/646ad154c37fbec184805b9b36626b7e8782a16c/crates/ty_python_semantic/src/types/ide_support.rs#L154-L178
